src/main/java/org/alfresco/service/namespace/QName.java

```diff
@@ -203,19 +195,42 @@ public static QName createQNameWithValidLocalName(final String namespaceURI, Str
      */
     public static String createValidLocalName(String name)
     {
-        // Validate length
-        if (name == null || name.length() == 0)
-        {
-            throw new IllegalArgumentException("Local name cannot be null or empty.");
-        }
+        checkLocalName(name);
         if (name.length() > MAX_LENGTH)
         {
             name = name.substring(0, MAX_LENGTH);
         }
 
         return name;
     }
-    
+
+    /**
+     * Condition to check string name provided for QName
+     *
+     * @param name  name to create valid local name from
+     * @throws InvalidQNameException
+     */
+    public static void checkLocalName(String name)
```

Comment:
It may be better to make this method package private. I'm not sure it can be used anywhere outside this class.

Reply:
Method converted to `private`.